---
title: "1.1_Differential_Expression_Analysis"
author: "Andrew Davis"
date: "12/2/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Differential Expression Analysis

John Blischak

Genes = features
Phenotype = variable of interest, i.e. treatment, genotype, age, etc
Expression levels (statistical models test differences in these measurements)

Why do large high throughput differential expression analysis?
novelty (identify new genes that change), context, systems

Caveats: measurements are relative, not absolute
Statistical methods cannot rescue a poorly designed study

Toy dataset:
Bioconductor package "breastCancerVDX"
344 patients: 209 ER+, 135 ER-


Bioconductor: CLL
22 patients 8 stable, 14 progressive

Expression matrix (x)
Feature data (f) - feature attributes (same # of rows as expression matrix)
Phenotype data (p) - describes each of the samples in the study (# of rows = the number of columns in the expression matrix)

boxplot(y-axis ~ x-axis, main = "title")

# Load dependencies:

```{r}
BiocManager::install("CLL")
library("CLL")
```




# Subset:
It's difficult to subset your dataset if you keep your 3 tables separately
To make analysis easier, Bioconductor provides useful classes - object oriented programming

class - defines a structure to hold complex data
object - a specific instance of a class
methods - functions that work on a specific class
getters/accessors - Get data stored in an object
setters/ - Modify data stored in an object

# Load package
library(Biobase)

# Create ExpressionSet object
eset <- ExpressionSet(assayData = x,
                      phenoData = AnnotatedDataFrame(p),
                      featureData = AnnotatedDataFrame(f))

# Accessing parts of an ExpressionSet:
Expression matrix
x <- exprs(eset)

Feature data
f <- fData(eset)

Phenotype data
p <- pData(eset)


```{r}
data(sCLLex)

BiocManager::install("annotate")
library(annotate)
BiocManager::install("hgu95av2.db")
library(hgu95av2.db)
ID <- featureNames(sCLLex)
Symbol <- getSYMBOL(ID,"hgu95av2.db")
fData(sCLLex) <- data.frame(Symbol=Symbol)


# Expression matrix
x <- exprs(sCLLex)

# Feature data
f <- fData(sCLLex)

# Phenotype data
p <- pData(sCLLex)

```


```{r}
boxplot(x[1,] ~ p[,"Disease"], main = f[1,"Symbol"])
```

# Subset with 3 separate objects:

Subset the first 10 samples and the first 1000 genes:
x_sub <- x[1000, 1:10]
f_sub <- f[1000, ]
p_sub <- p[1:10, ]

Subset with an ExpressionSet object:

eset_sub <- eset[1000, 1:10]

nrow(exprs(eset_sub)) == nrow(fData(eset_sub))

[1] TRUE

ncol(exprs(eset_sub)) == nrow(pData(eset_sub))

[1] TRUE

